index.js

```diff
@@ -3,30 +3,153 @@ import Fiber from 'fibers'
 
 const SYNC_COMMANDS = ['domain', '_events', '_maxListeners', 'setMaxListeners', 'emit',
     'addListener', 'on', 'once', 'removeListener', 'removeAllListeners', 'listeners']
-const NOOP = function () {}
 
-let wrapCommand = function (instance, hooks) {
-    Object.keys(Object.getPrototypeOf(instance)).forEach((commandName) => {
-        if (SYNC_COMMANDS.indexOf(commandName) > -1) {
-            return
+let commandIsRunning = false
+
+/**
+ * helper method to execute a row of hooks with certain parameters
+ * @param  {Function|Function[]} hooks  list of hooks
+ * @param  {Object[]} args  list of parameter for hook functions
+ * @return {Promise}  promise that gets resolved once all hooks finished running
+ */
+let executeHooksWithArgs = (hooks, args) => {
+    /**
+     * make sure hooks are an array of functions
+     */
+    if (typeof hooks === 'function') {
+        hooks = [hooks]
+    }
+
+    /**
+     * make sure args is an array since we are calling apply
+     */
+    if (!Array.isArray(args)) {
+        args = [args]
+    }
+
+    hooks = hooks.map((hook) => new Promise((resolve) => {
+        let _commandIsRunning = commandIsRunning
+        try {
+            /**
+             * after command hooks require additional Fiber environment
+             */
+            return Fiber(() => {
+                commandIsRunning = true
+                resolve(hook.apply(null, args))
+                commandIsRunning = _commandIsRunning
+            }).run()
+        } catch (e) {
+            // here could be your error message
```

Comment:
Could we log this to the console, or something, for now? I had some interesting errors swallowed up here.